### PR TITLE
Feature/cypress codecov

### PR DESCRIPTION
**Pull request by jeffvg**
#### Changes proposed in this Pull Request:
- add @cypress/code-coverage, @cypress/instrument-cra and babel-plugin-istanbul
- add nyc cypress-coverage for dir name/location
- add imports for cypress codecov
- adjust ci config to upload cypress codecov report
- include start command only for cypress
- add additional test for empty state on overlays and clean up overlay click to dismiss bottom-sheet
- two report files merged by codecov (no configuration needed)

https://codecov.io/gh/pxblue/react-design-patterns/tree/feature%2Fcypress-codecov/src

image
 

#### To Test:
-

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -79,6 +83,9 @@ jobs:
             - codecov/upload:
                 file: './coverage/clover.xml'
                 token: 74ebd8f2-0b51-4873-972a-82c2b3509b37
+            - codecov/upload:
+                file: './cypress-coverage/clover.xml'
+                token: 74ebd8f2-0b51-4873-972a-82c2b3509b37
```
Comment (joebochill) [inline .circleci/config.yml:88]: With two upload steps here is codecov smart enough to know that these are different files and not just override the first one with the second?

**Review by joebochill — APPROVED** [review]
